From ed0734dcf44a5eef4b17ebc5237a0b2589137943 Mon Sep 17 00:00:00 2001
From: Jiaqing Zhao <jiaqing.zhao@intel.com>
Date: Sun, 7 May 2023 03:32:58 +0800
Subject: [PATCH] OvmfPkg: add PcdPciSerialParameters for CH382 PCIe serial
 adapter

This patch adds a pre-defined PcdPciSerialParameters for CH382 PCIe
serial adapter to make it works with PciSioSerialDxe driver.

CH382(1c00:3253) is a dual port 16850-compatible PCIe serial adapter
with single PCI function, its port 0 is at io_base+0xc0, and port 1 is
at io_base+0xc8.

Signed-off-by: Jiaqing Zhao <jiaqing.zhao@intel.com>
---
 OvmfPkg/OvmfPkgX64.dsc | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/OvmfPkg/OvmfPkgX64.dsc b/OvmfPkg/OvmfPkgX64.dsc
index e3c64456df..821af0407f 100644
--- a/OvmfPkg/OvmfPkgX64.dsc
+++ b/OvmfPkg/OvmfPkgX64.dsc
@@ -625,6 +625,17 @@
   # Point to the MdeModulePkg/Application/UiApp/UiApp.inf
   gEfiMdeModulePkgTokenSpaceGuid.PcdBootManagerMenuFile|{ 0x21, 0xaa, 0x2c, 0x46, 0x14, 0x76, 0x03, 0x45, 0x83, 0x6e, 0x8a, 0xb6, 0xf4, 0x66, 0x23, 0x31 }
 
+  #
+  # Patch for CH382 Dual Port PCIe Serial Adapter
+  # VendorId    DeviceId    ClockRate               Offset (Port0: 0x0c0, Port1: 0x0c8)             Bar   Stride RxFifo      TxFifo      Rsvd
+  # 0x00, 0x1c, 0x53, 0x32, 0x00, 0x00, 0x00, 0x00, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,  0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+  # 0x00, 0x1c, 0x53, 0x32, 0x00, 0x00, 0x00, 0x00, 0xc8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,  0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+  # 0xff, 0xff
+  #
+!if $(SERIAL_CH382)
+  gEfiMdeModulePkgTokenSpaceGuid.PcdPciSerialParameters|{0x00, 0x1c, 0x53, 0x32, 0x00, 0x00, 0x00, 0x00, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1c, 0x53, 0x32, 0x00, 0x00, 0x00, 0x00, 0xc8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0xff}
+!endif
+
 ################################################################################
 #
 # Pcd Dynamic Section - list of all EDK II PCD Entries defined by this Platform
-- 
2.39.2

